File: LeetCode/1200/main.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        ans = []
        min_diff = float('inf')

        arr.sort()

        for i in range(len(arr) - 1) :
            min_diff = min(min_diff, abs(arr[i] - arr[i+1]))

        for i in range(len(arr) - 1) :
            if(abs(arr[i] - arr[i+1]) == min_diff) :
                ans.append([arr[i], arr[i+1]])

        return ans

File: LeetCode/1200/main.cpp
class Solution {
public:
  vector<vector<int>> minimumAbsDifference(vector<int> &arr) {
    vector<vector<int>> ans;
    int min_diff = INT_MAX;

    sort(arr.begin(), arr.end());

    for (int i = 0; i < arr.size() - 1; i++) {
      if (abs(arr[i] - arr[i + 1]) < min_diff) {
        min_diff = abs(arr[i] - arr[i + 1]);
      }
    }

    for (int i = 0; i < arr.size() - 1; i++) {
      if (abs(arr[i] - arr[i + 1]) == min_diff) {
        ans.emplace_back(vector<int>{arr[i], arr[i + 1]});
      }
    }

    return ans;
  }
};